### ISSUE TITLE
Upgrading peer dependencies (RN + RNSVG)

### ISSUE THREAD

**Issue by vsheyanov** [LovesWorking/react-native-react-query-devtools#7]
Is it possible (safe) to upgrade peer dependecies to:
```
"react-native": "~0.74.0",
"react-native-svg": "^15"
```
Right now the peer dependency is locked with previous versions of both packages.
I did an override in `package.json` to suppress the error during `npm i` but I'm wondering if there is something that should be aware of.

Thank you!

**Comment by GregPetropoulos** [comment]
I am running `expo 51` and having the same peer dep conflict. I tried patching by bringing the `react-native` dep up to version 74 and that failed. 

I also tried using the resolution block in my root `package.json` and that failed. So no love from DevToolsBubble for RN devs I guess.  🙃

**Comment by martinezguillaume** [comment]
Me too, error on install with react-native dep :( no love for me

**Comment by LovesWorking** [comment]
Thanks for the report. Can you please make a PR for this change? @martinezguillaume  @GregPetropoulos  @vsheyanov If none of you can. I can try to make the change tonight.

**Comment by GregPetropoulos** [comment]
@LovesWorking I have WIP PR, can you allow me permission to push a branch up I am getting a 403


<img width="1213" alt="Screenshot 2024-07-09 at 11 08 59 AM" src="https://github.com/LovesWorking/react-native-react-query-devtools/assets/68525155/cde10cb9-9ed4-46b6-81f8-a23a977ac27e">

**Comment by toppsdown** [comment]
+1 for same problem

**Comment by LovesWorking** [comment]
> @LovesWorking I have WIP PR, can you allow me permission to push a branch up I am getting a 403
> 
> <img alt="Screenshot 2024-07-09 at 11 08 59 AM" width="1213" src="https://private-user-images.githubusercontent.com/68525155/347013204-cde10cb9-9ed4-46b6-81f8-a23a977ac27e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2NDgzNzksIm5iZiI6MTcyMDY0ODA3OSwicGF0aCI6Ii82ODUyNTE1NS8zNDcwMTMyMDQtY2RlMTBjYjktOWVkNC00NmI2LTgxZjgtYTIzYTk3N2FjMjdlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEwVDIxNDc1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjMTJkNjNhYTJmNzYyYTBhMmE4YTNjODBjNDEyOTI5NjBjODkzNzY5ZGVhMjlmN2QwZDU2YjJkNGJjMGViNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6JowHtCZ16zzrvMFMPIxfeO9XjdgrKpRAbu78Yo0UMg">

Invite sent. Once accepted you should be able to make a pull request. Thank you!

**Comment by GregPetropoulos** [comment]
> > @LovesWorking I have WIP PR, can you allow me permission to push a branch up I am getting a 403
> > <img alt="Screenshot 2024-07-09 at 11 08 59 AM" width="1213" src="https://private-user-images.githubusercontent.com/68525155/347013204-cde10cb9-9ed4-46b6-81f8-a23a977ac27e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2NDgzNzksIm5iZiI6MTcyMDY0ODA3OSwicGF0aCI6Ii82ODUyNTE1NS8zNDcwMTMyMDQtY2RlMTBjYjktOWVkNC00NmI2LTgxZjgtYTIzYTk3N2FjMjdlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEwVDIxNDc1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjMTJkNjNhYTJmNzYyYTBhMmE4YTNjODBjNDEyOTI5NjBjODkzNzY5ZGVhMjlmN2QwZDU2YjJkNGJjMGViNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6JowHtCZ16zzrvMFMPIxfeO9XjdgrKpRAbu78Yo0UMg">
> 
> Invite sent. Once accepted you should be able to make a pull request. Thank you!

@LovesWorking I am not seeing an invite in my email, can you resend it?

**Comment by LovesWorking** [comment]
> Is it possible (safe) to upgrade peer dependecies to:
> 
> ```
> "react-native": "~0.74.0",
> "react-native-svg": "^15"
> ```
> 
> Right now the peer dependency is locked with previous versions of both packages. I did an override in `package.json` to suppress the error during `npm i` but I'm wondering if there is something that should be aware of.
> 
> Thank you!

Try version 1.2.0 and let me know if it all works. Thanks.

**Comment by GregPetropoulos** [comment]
TLDR; Can't have a `div` for React Native, we use `Views`.

@LovesWorking I appreciate your feedback. I don't see the correct changes in the lock file of the approved PR for the peerDeps merged, looks like someone poorly entered in the version number but that's a topic for another day.

The peerDeps don't have much to do with the bug I am getting and most likely went unnoticed for a while. 

In fact, I dug deeper into the tanstack library and documentation I am pretty sure the `ReactQueryDevtools` is not supported for mobile devices. I haven't checked `react-native-webview` though.

In my screenshot on the left you can see the error I am getting in the simulator (I checked real devices as well). 
Also, deep in the node modules I circled the div causing the issue and in my expo go template app. 
On the right you can see the `ReactQueryDevtools` are turned on. 🤷‍♂️ 

<img width="1440" alt="Screenshot 2024-07-11 at 4 32 51 PM" src="https://github.com/LovesWorking/react-native-react-query-devtools/assets/68525155/a0696d48-1b94-496b-a0a3-b8f78dcaf302">

**Comment by martinezguillaume** [comment]
Sorry for the confusion, here is the [other pr](https://github.com/LovesWorking/react-native-react-query-devtools/pull/11) to fix the deps

@GregPetropoulos yes react-query-devtools is not supported on react-native, for now you have to solution :
- this lib 🚀
- [this lib](https://github.com/expo/dev-plugins/tree/main/packages/react-query) if you are using expo

**Comment by martinezguillaume** [comment]
Like you want ! We can close this issue and reopen another one
